### PR TITLE
Fixes for dolt_tests

### PR DESCRIPTION
**Pull request by NathanGabrielson**
Couple of fixes for `dolt_tests`:
1. Allow comparison of int64 values, which enables running aggregate queries like `sum` and `avg`, for instance.
2. Can check if values are/are not null

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@NathanGabrielson __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>25c8247</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>25c8247</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/env/actions/test_table_helpers.go [hunk 1]
```diff
@@ -295,15 +310,37 @@ func compareDecimals(comparison string, expectedValue, realValue decimal.Decimal
...
+	case "==":
+		if actualValue != nil {
+			return fmt.Sprintf("Assertion failed: %s equal to NULL, got %v", assertionType, actualValue)
+		}
+	case "!=":
+		if actualValue == nil {
+			return fmt.Sprintf("Assertion failed: %s not equal to NULL, got NULL", assertionType)
+		}
+	default:
+		return fmt.Sprintf("%s is not a valid assertion type", comparison)
```
Comment (macneale4) [inline go/libraries/doltcore/env/actions/test_table_helpers.go]:
```suggestion
		return fmt.Sprintf("%s is not a valid comparison for NULL values", comparison)
```

File: go/libraries/doltcore/sqle/dtablefunctions/dolt_test_run.go [hunk 2]
```diff
@@ -337,5 +342,9 @@ func validateQuery(ctx *sql.Context, catalog sql.Catalog, query string) (string,
 	} else if isWrite {
 		return "Cannot execute write queries", nil
 	}
+
+	if strings.Contains(query, "dolt_test_run(") {
```
Comment (macneale4) [inline go/libraries/doltcore/sqle/dtablefunctions/dolt_test_run.go]: Perform a case insensitive match here.

File: go/libraries/doltcore/sqle/enginetest/dolt_queries_test_table.go [hunk 3]
```diff
@@ -293,6 +307,27 @@ var DoltTestRunFunctionScripts = []queries.ScriptTest{
...
+				"('simple null string equality', 'SELECT t FROM numbers', 'expected_single_value', '=='), " +
+				"('simple null inequality', 'SELECT i FROM numbers', 'expected_single_value', '!=')",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM dolt_test_run('*')",
+				Expected: []sql.Row{
+					{"simple null inequality", "", "SELECT i FROM numbers", "FAIL", "Assertion failed: expected_single_value not equal to NULL, got NULL"},
+					{"simple null int equality", "", "SELECT i FROM numbers", "PASS", ""},
+					{"simple null string equality", "", "SELECT t FROM numbers", "PASS", ""},
```
Comment (macneale4) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries_test_table.go:328]: Looks like you are missing a test where you are looking for a NULL value, but the test return fails. Would be good to ensure the message is correct.

**Comment by coffeegoddd** [comment]
@NathanGabrielson __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>7b676b8</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>7b676b8</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>
